lib/services/plugins-service.ts

```diff
@@ -0,0 +1,190 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import shelljs = require("shelljs");
+import semver = require("semver");
+import Future = require("fibers/future");
+import constants = require("./../constants");
+let xmlmerge = require("xmlmerge-js");
+
+export class PluginsService implements IPluginsService {
+	private static INSTALL_COMMAND_NAME = "install";
+	private static UNINSTALL_COMMAND_NAME = "uninstall";
+	
+	constructor(private $npm: INodePackageManager,
+		private $fs: IFileSystem,
+		private $projectData: IProjectData,
+		private $platformsData: IPlatformsData,
+		private $projectDataService: IProjectDataService,
+		private $childProcess: IChildProcess,
+		private $options: IOptions,
+		private $logger: ILogger,
+		private $errors: IErrors) { }
+		
+	public add(plugin: string): IFuture<void> {
+		return (() => {
+			let pluginName = this.executeNpmCommand(PluginsService.INSTALL_COMMAND_NAME, plugin).wait();
+			let nodeModuleData = this.getNodeModuleData(pluginName);
+			if(!nodeModuleData.isPlugin) {
+				// We should remove already downloaded plugin and show an error message
+				this.executeNpmCommand(PluginsService.UNINSTALL_COMMAND_NAME, pluginName).wait();
+				this.$errors.failWithoutHelp(`The specified plugin ${plugin} is not a valid NativeScript plugin. Ensure that the plugin contains nativescript key in package.json file and try again.`);
```

Comment:
${plugin} is not a valid NativeScript plugin. Verify that the plugin package.json file contains a nativescript key and try again.